examples/parse_args.h

```diff
@@ -0,0 +1,175 @@
+//
+// Copyright (c) 2024 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+#pragma once
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <strings.h>
+
+#include "zenoh.h"
+
+#define COMMON_HELP \
+    "\
+        -c <CONFIG> (optional, string): The path to a configuration file for the session. If this option isn't passed, the default configuration will be used.\n\
+        -m <MODE> (optional, string, default='peer'): JSON-serialized string of the zenoh session mode. [possible values: 'peer', 'client', 'router']\n\
+        -e <CONNECT> (optional, string): JSON-serialized list of locators to connect to. If none are given, endpoints will be discovered through multicast-scouting if it is enabled.\n\
+        -l <LISTEN> (optional, string): JSON-serialized list of locators to listen on. If none are given, the default configuration will be used.\n\
+        --no-multicast-scouting (optional): By default zenohd replies to multicast scouting messages for being discovered by peers and clients. This option disables this feature.\n\
+"
+
+/**
+ * Parse an option of format `-f`, `--flag`, `-f <value>` or `--flag <value>` from `argv`. If found, the option and its
+ * eventual value are each replaced by an empty string in `argv`
+ * @param argc: argc passed from `main` function
+ * @param argv: argv passed from `main` function
+ * @param opt: option to parse (without `-` or `--` prefix)
+ * @param opt_has_value: if true, the option is of format `-f <value>` or `--flag <value>` and `value` will be returned
+ * if found, else an error message is printed and program will exit. If false, option has no value and a non-null
+ * pointer will be returned if option is found.
+ * @returns NULL if option was not found, else a non-null value depending on if `opt_has_value`.
+ */
+char* parse_opt(int argc, char** argv, char* opt, bool opt_has_value) {
+    size_t optlen = strlen(opt);
+    for (int i = 1; i < argc; i++) {
+        size_t len = strlen(argv[i]);
+        if (len >= 2) {
+            if (optlen == 1) {
+                if (argv[i][0] == '-' && argv[i][1] == opt[0]) {
+                    argv[i][0] = '\0';
+                    if (!opt_has_value) {
+                        return opt;
+                    } else if (i + 1 < argc && strlen(argv[i + 1]) > 0) {
+                        char* buf = (char*)malloc(strlen(argv[i + 1]));
+                        strcpy(buf, argv[i + 1]);
```

Comment:
Better use strncpy.
And also consider usage of `strdup` (I am not sure about windows)